cost-analyzer/templates/forecasting-deployment.yaml

```diff
@@ -0,0 +1,102 @@
+{{- if .Values.forecasting.enabled }}
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "forecasting.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{ include "forecasting.commonLabels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{ include "forecasting.selectorLabels" . | nindent 6 }}
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: forecasting
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        app: forecasting
+      {{- with .Values.global.podAnnotations}}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+      restartPolicy: Always
+      containers:
+        - name: forecasting
+          {{- if .Values.kubecostModel }}
+          {{- if .Values.kubecostModel.openSourceOnly }}
+          {{- fail "Kubecost Forecasting cannot be used with open source only" }}
+          {{- else if .Values.forecasting.fullImageName }}
+          image: {{ .Values.forecasting.fullImageName }}
+          {{- else if .Values.kubecostModel.fullImageName }}
+          image: {{ .Values.kubecostModel.fullImageName }}
+          {{- else if .Values.imageVersion }}
+          image: {{ .Values.kubecostModel.image }}:{{ .Values.imageVersion }}
+          {{- else }}
+          image: {{ .Values.kubecostModel.image }}:prod-{{ $.Chart.AppVersion }}
+          {{ end }}
+          {{- else }}
+          image: gcr.io/kubecost1/cost-model:prod-{{ $.Chart.AppVersion }}
+          {{ end }}
+          args: ["forecasting"]
+          {{/*
+          fix readinessProbe
+
+          readinessProbe:
+            httpGet:
+              path: /healthz
+              port: 5000
+            initialDelaySeconds: 10
+            periodSeconds: 5
+            failureThreshold: 200
+          */}}
+          imagePullPolicy: Always
+          ports:
+            - name: tcp-api
+              containerPort: 5000
+              protocol: TCP
+          resources:
+            {{- toYaml .Values.forecasting.resources | nindent 12 }}
+          volumeMounts:
+          env:
+            - name: CONFIG_PATH
+              value: /var/configs/
+            - name: KCM_BASE_URL
+              value: "https://aggregator.nightly.kubecost.xyz/model"
+            - name: MODEL_STORAGE_PATH
+              value: "/tmp/localrun/models"
```

Comment:
I assume you're writing your models to disk at this location? How large are these models? If they're sufficiently large we may want to consider creating a PersistentVolume for them.

Reply:
They are starting out relatively simple, and we're focusing on functionality first over perfect stability of saved models, so this isn't a major initial concern. This is an excellent general callout, though! We have to follow up here when the models get more sophisticated and the component has been proven out more.

Reply:
Ditto. Also note because this does not yet adopt global securityContext, writing to `/tmp` may be disallowed with `readOnlyRootFilesystem` set to `true`.

Reply:
With r/o root fs, I added an empty dir volume to make this work correctly. In the future we'll want to consider the PV question more closely and ideally make the path configurable.

Reply:
That works, but there should probably be a size limit set for this. emptyDir's default backing medium is memory. Do we know how large these models are, per Thomas' question?

Reply:
Currently, the models are very small json files, generally 1-2 Mb for all of them

Reply:
Thanks Chip, I didn't know that -- set sizeLimit on the emptydir to 500Mi to start